Mathlib/NumberTheory/NumberField/Embeddings.lean

```diff
@@ -1005,6 +1007,140 @@ lemma card_eq_card_isUnramifiedIn [NumberField k] [IsGalois k K] :
+/-- The normed field structure of a number field coming from the norm asociated to
+an infinite place. -/
+noncomputable def normedField : NormedField K :=
+  NormedField.induced _ _ v.embedding v.embedding.injective
+
+/-- The normed division ring structure of a number field coming from the norm associated
+to an infinite place. -/
+noncomputable def normedDivisionRing : NormedDivisionRing K :=
+  v.normedField.toNormedDivisionRing
+
+/-- The uniform structure of a number field at an infinite place. -/
+noncomputable def uniformSpace : UniformSpace K :=
+  v.normedField.toUniformSpace
+
+/-- The uniform additive group structure of a number field induced by an infinite place. -/
+theorem uniformAddGroup : @UniformAddGroup K v.uniformSpace _ :=
+  letI := v.normedField
+  inferInstance
+
+/-- The topology of a number field infuced by an infinite place. -/
+noncomputable def topologicalSpace : TopologicalSpace K :=
+  v.uniformSpace.toTopologicalSpace
+
+/-- The topological division ring structure of a number field induced by an infinite place. -/
+theorem topologicalDivisionRing :
+    @TopologicalDivisionRing K _ v.topologicalSpace :=
+  v.normedDivisionRing.to_topologicalDivisionRing
+
+/-- The topological ring structure of a number field induced by an infinite place. -/
+theorem topologicalRing : @TopologicalRing K v.topologicalSpace _ :=
+  @TopologicalDivisionRing.toTopologicalRing _ _ v.topologicalSpace
+    v.topologicalDivisionRing
+
+/-- The embedding associated to an infinite place is a uniform embedding. -/
+theorem uniformEmbedding : @UniformEmbedding _ _ v.uniformSpace _ v.embedding := by
+  rw [@uniformEmbedding_iff, @uniformInducing_iff_uniformSpace]
+  exact ⟨rfl, v.embedding.injective⟩
+
+/-- The embedding associated to an infinite palce is uniform inducing. -/
+theorem uniformInducing : @UniformInducing _ _ v.uniformSpace _ v.embedding :=
+  @UniformEmbedding.toUniformInducing _ _ v.uniformSpace _ _ v.uniformEmbedding
+
+/-- The embedding associated to an infinite place of a number field is an isometry. -/
+theorem isometry : @Isometry _ _ v.normedField.toPseudoEMetricSpace _ (v.embedding) :=
+  @UniformEmbedding.to_isometry _ _ v.uniformSpace _ _ v.uniformEmbedding
+
+/-- The embedding associated to an infinite place of a number field is uniform continuous. -/
+theorem uniformContinuous : @UniformContinuous _ _ v.uniformSpace _ v.embedding :=
+  @UniformInducing.uniformContinuous _ _ v.uniformSpace _ _ v.uniformInducing
+
+/-- The embedding associated to an infinite place of a number field is continuous. -/
+theorem continuous : @Continuous _ _ v.topologicalSpace _ v.embedding :=
+  @UniformContinuous.continuous _ _ v.uniformSpace _ _ v.uniformContinuous
+
+/-- The uniform structure induced by an infinite place of a number field defines a
+completable topological field. -/
+theorem completableTopField : @CompletableTopField K _ v.uniformSpace :=
+  @UniformSpace.comap_completableTopField _ _ _ _ _ _ v.normedField.instT0Space _
```

Comment:
Similarly, I don't think these are needed. The point of type class inference is to find these for you.

Reply:
The first four are gone now. I have left `completableTopField` separate there since it has some non-trivial proof going on in `comap_completableTopField`, which itself is included elsewhere in this PR. But this could also be removed with a `haveI` in the `NormedField v.completion` instance further down, if preferred.

Reply:
`comap_completableTopField` could be an instance. Then this is not necessary.

Reply:
Makes sense, fixed that.